File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/domain/sample/AbstractMailMessage.java
package com.coupang.springframework.data.requery.domain.sample;

import com.coupang.springframework.data.requery.domain.AbstractPersistable;
import io.requery.*;

/**
 * AbstractMailMessage
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
@Entity
public abstract class AbstractMailMessage extends AbstractPersistable<Long> {

    @Key
    @Generated
    protected Long id;

    @ForeignKey
    @OneToOne
    protected AbstractMailSender mailSender;

    protected String content;

    private static final long serialVersionUID = -6401169833123419204L;
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractMailMessage.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.*

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractMailMessage
 * @author debop
 * @since 18. 5. 30
 */
@Entity
abstract class AbstractMailMessage: AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    abstract override val id: Long?

    @get:OneToOne(cascade = [CascadeAction.SAVE, CascadeAction.DELETE])
    @get:ForeignKey
    abstract var mailSender: AbstractMailSender?

    abstract var content: String?
}